src/v/redpanda/application.cc

```diff
@@ -1370,6 +1333,79 @@ void application::start_redpanda(::stop_signal& app_signal) {
 
     storage.invoke_on_all(&storage::api::start).get();
 
+    syschecks::systemd_message("Starting internal RPC bootstrap service").get();
+    _rpc
+      .invoke_on_all([this](rpc::rpc_server& s) {
+          std::vector<std::unique_ptr<rpc::service>> bootstrap_service;
+          bootstrap_service.push_back(
+            std::make_unique<cluster::bootstrap_service>(
+              _scheduling_groups.cluster_sg(),
+              smp_service_groups.cluster_smp_sg()));
+          s.add_services(std::move(bootstrap_service));
+      })
+      .get();
+    _rpc.invoke_on_all(&rpc::rpc_server::start).get();
+    vlog(
+      _log.info,
+      "Started RPC server listening at {}",
+      config::node().rpc_server());
+}
+
+void application::wire_up_and_start(::stop_signal& app_signal, bool test_mode) {
+    wire_up_bootstrap_services();
+    start_bootstrap_services();
+
+    // Begin the cluster discovery manager so we can determine our initial node
+    // ID. A valid node ID is required before we can initialize the rest of our
+    // subsystems.
+    const auto& node_uuid = storage.local().node_uuid();
+    cluster::cluster_discovery cd(node_uuid);
+    auto node_id = cd.determine_node_id().get();
```

Comment:
is this the point in start-up that enforces the dependency between storage/rpc/discovery and the rest of the services?

Reply:
Yeah, in the node ID assignment PR, `determine_node_id()` may end up sending out some RPCs, so we'll end up blocking here before proceeding (because we can't, because we need to know our node ID to start up Raft, etc).